### DISCUSSION TITLE
Support Drizzle's `.query` api in AstroDB (a.k.a. relational queries)

### DISCUSSION THREAD

**Discussion by nicojs** [withastro/roadmap#992] [Proposal]
# Summary

The docs mention that [astrodb uses Drizzle for querieing your data](https://docs.astro.build/en/guides/astro-db/#query-your-database). However, only the `.select()` API is documented and usable from `astro/db`, which is the "query-builder API". The Drizzle website mentions the more powerful [`.query` api](https://orm.drizzle.team/docs/rqb). 

Could we also expose the `.query` API in astrodb?

The `.query` API seems to be also known as "relational queries".

# Background & Motivation

I've asked around in the Astro Discord and @delucis let me know that originally the Drizzle `.query` API was experimental. Since then, however, it is no longer experimental and seems to be the preferred way to query (since it is the first page in the Drizzle docs).

Note that the `.query` API allows for much easier querying, since you don't have to de-duplicate the result of a query with custom `.reduce` functions. It makes Drizzle an actual ORM, rather than a advanced query builder.

# Goals

Be able to use the full [`.query` api](https://orm.drizzle.team/docs/rqb). 

# Example

```ts
const result = await db.query.users.findMany({
  with: {
    posts: true      
  },
});
```

```js
[{
  id: 10,
  name: "Dan",
  posts: [
    {
      id: 1,
      content: "SQL is awesome",
      authorId: 10,
    },
    {
      id: 2,
      content: "But check relational queries",
      authorId: 10,
    }
  ]
}]
```

**Comment by florian-lefebvre** [comment]
Astro DB has been deprecated
